src/libmongoc/tests/test-mongoc-hello.c

```diff
@@ -0,0 +1,164 @@
+   ASSERT (request);
+   mock_server_replies_simple (request,
+                               "{'ok': 1, 'isWritablePrimary': true }");
+   request_destroy (request);
+
+   ret = future_get_bool (future);
+   ASSERT (ret);
+   future_destroy (future);
+
+   bson_destroy (ping);
+   mongoc_client_destroy (client);
+   mock_server_destroy (server);
+
+   /* If we made it here, everything went just fine: */
+   return true;
+}
+
+void
+test_mongoc_hello ()
+{
+   /* Be sure the test didn't pass when it should /not/ have: */
+   ASSERT (_test_mongoc_should_check_hello () && _test_mongoc_hello_impl ());
+}
+
+bool
+_test_mongoc_hello_client_pool_impl ()
+{
+   mongoc_client_t
+      *cl3; /* the command client, which we'll push back into the pool */
+   bson_t *cmd;
+   bson_error_t err;
+   bool cmd_result;
+
+   mongoc_server_api_t *requested_server_api = NULL;
+
+   mongoc_uri_t *test_uri = test_framework_get_uri ();
+
+   /* The test_framework_client_pool_new_from_uri() function checks that the
+   requeted API can be set; if NULL is passed, it uses the default server API
+   via a call to test_framework_get_default_server_api(), which here will
+   basically come from an environment variable: */
+   mongoc_client_pool_t *pool =
+      test_framework_client_pool_new_from_uri (test_uri, requested_server_api);
+
+   BSON_ASSERT (pool);
+   test_framework_set_pool_ssl_opts (pool);
+
+   /* Now, set up a couple of clients, and a ping: */
+   (void) mongoc_client_pool_pop (pool);
+   (void) mongoc_client_pool_pop (pool);
+
+   cmd = BCON_NEW ("ping", BCON_INT32 (1));
+
+   cl3 = mongoc_client_pool_pop (pool);
+
+   cmd_result =
+      mongoc_client_command_simple (cl3, "admin", cmd, NULL, NULL, &err);
+
+   ASSERT_OR_PRINT (cmd_result, err);
+
+   mongoc_client_pool_push (pool, cl3);
+
+   // Tidy up:
+   mongoc_client_pool_destroy (pool);
+   mongoc_uri_destroy (test_uri);
+   mongoc_server_api_destroy (requested_server_api);
+
+   /* If we're here, things went ok: */
+   return true;
+}
+
+/* Wrap the underlying test to be sure that it does not /accidentally/ pass when
+it shouldn't. For example, here we want to run the same test whether or not the
+target wire protocol version is supposed to support OP_MSG, but it should
+already FAIL before it reaches this point; we are performing a final check,
+which is to say that it should never pass when the right features aren't
+present: */
+void
+test_mongoc_hello_client_pool ()
+{
+   ASSERT (_test_mongoc_should_check_hello () &&
+           _test_mongoc_hello_client_pool_impl ());
+}
+
+void
+test_hello_install (TestSuite *suite)
+{
+   TestSuite_AddMockServerTest (
+      suite,
+      "/hello",
+      test_mongoc_hello,
+      test_framework_skip_if_max_wire_version_less_than_5);
+   TestSuite_AddMockServerTest (
+      suite,
+      "/hello/client_pool",
+      test_mongoc_hello_client_pool,
+      test_framework_skip_if_max_wire_version_less_than_5);
```

Comment:
`test_framework_skip_if_max_wire_version_less_than_5` checks the wire version of a live server running for tests. Since these tests are using the mock server, the `test_framework_skip_if_max_wire_version_less_than_5` can be removed.

Reply:
Ah, I meant to do this! Thank you for catching it.